RealmSwift/Results.swift

```diff
@@ -353,19 +353,74 @@ extension AnyRealmValue: AddableType {}
+     modification, or deletion for all object properties and the properties of
+     any nested, linked objects. If a key path or key paths are provided,
+     then the block will be called for changes which occur only on the
+     provided key paths. For example, if:
+     ```swift
+     class Dog: Object {
+         @objc dynamic var name: String = ""
+         @objc dynamic var age: Int = 1
+         let toys = List<Toy>()
+     }
+     // ...
+     let dogs = realm.objects(Dog.self)
+
+     let token = dogs.observe(keyPaths: ["name"]) { changes in
+         switch changes {
+         case .initial(let dogs):
+            // ...
+         case .update:
+            // This case is hit:
+            // - after the token is intialized
+            // - when an element of the collections' name
+            // property is modified
+            // - when an element is inserted or deleted.
+            // This block is not triggered:
+            // - when any of the elements "age" values is modified
+         case .error:
+             // ...
+         }
+     }
+     ```
+     - If the above example observed the `["toys"]` key path, then any insertion,
+     deletion, or modification to the `toys` list for any element in the collection would trigger the block.
+     Any insertion or deletion to the `Dog` type collection being observed
+     would also trigger a notification.
+     - If the observed key path were `["toys.brand"]`, then any insertion or
+     deletion to the `toys` list of any collection element would trigger the block. Changes to the `brand` value
+     on any `Toy` that is linked to a `Dog` in this collection will trigger the block. Changes to any other value
+     on any `Toy` that is linked to a `Dog` in this collection would not trigger the block.
+     Any insertion or deletion to the `Dog` type collection being observed
+     would also trigger a notification.
+
+     - note: Multiple notification tokens on the same object which filter for
+     separate key paths *do not* filter exclusively. If one key path
+     change is satisified for one notification token, then all notification
+     token blocks for that object will execute.
+
+     If no queue is given, notifications are delivered via the standard run
+     loop, and so can't be delivered while the run loop is blocked by other
+     activity. If a queue is given, notifications are delivered to that queue
+     instead. When notifications can't be delivered instantly, multiple
+     notifications may be coalesced into a single notification.
+
      You must retain the returned token for as long as you want updates to be sent to the block. To stop receiving
      updates, call `invalidate()` on the token.
-
      - warning: This method cannot be called during a write transaction, or when the containing Realm is read-only.
-
+     - parameter keyPaths: The element type properties which trigger the block to
```

Comment:
This wording is a bit awkward. "String keyPaths to an element's property, e.g., <example here>" or something of that ilk.